src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            Node<T> currentNode = new Node<>(tail, value, null);
+            tail.next = currentNode;
+            tail = currentNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            Node<T> currentNode = new Node<>(null, value, head);
+            head.prev = currentNode;
+            head = currentNode;
+        } else {
+            indexValidation(index);
+            Node<T> currentNode = nodeFromIndex(index);
+            Node<T> node = new Node<>(currentNode.prev, value, currentNode);
+            currentNode.prev.next = node;
+            currentNode.prev = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
```

Comment:
Every time you call `indexValidation()`, on the next line you have `nodeFromIndex()`, so you can just move `indexValidation()` call at the start of the `nodeFromIndex()`, this will will improve readability:
```
private Node<T> nodeFromIndex(int index) {
    indexValidation(index);
    //rest of your code...
}
```